### ISSUE TITLE
Exception appears when Narrator is active and user tries to add row after removing of another row.

### ISSUE THREAD

**Issue by SergeySmirnov-Akvelon** [dotnet/winforms#4228]
* .NET Core Version: 
5.0.100-rc.2.20479.15

* Have you experienced this same bug with .NET Framework?: 
No

**Minimal repro:**

1. Create the application with a DataGridView
2. Add a several rows to the DataGridView.
3. Or just use the attached project ([4228.zip](https://github.com/dotnet/winforms/files/5536419/4228.zip))
4. Run the application.
5. Start editing the last row (row to add a new row)
6. Delete the last row with data
7. Start editing the last row (row to add a new row)

![Issue-4228](https://user-images.githubusercontent.com/23376742/99064105-89234e00-25b6-11eb-9cfe-32510b3e7604.gif)

**Actual behavior:**
The error is displayed.
```
System.Windows.Forms.dll!System.Windows.Forms.AccessibleObject.Interop.UiaCore.IRawElementProviderFragment.FragmentRoot.get() Line 657	C#
[External Code]	
System.Windows.Forms.dll!System.Windows.Forms.Application.ComponentManager.Interop.Mso.IMsoComponentManager.FPushMessageLoop(System.UIntPtr dwComponentID, Interop.Mso.msoloop uReason, void* pvLoopData) Line 345	C#
System.Windows.Forms.dll!System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Interop.Mso.msoloop reason, System.Windows.Forms.ApplicationContext context) Line 1111	C#
System.Windows.Forms.dll!System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Interop.Mso.msoloop reason, System.Windows.Forms.ApplicationContext context) Line 975	C#
System.Windows.Forms.dll!System.Windows.Forms.Application.Run(System.Windows.Forms.Form mainForm) Line 1198	C#
WindowsFormsApp25.dll!WindowsFormsApp25.Program.Main() Line 20	C#
```

**Expected behavior:**
The error should not be displayed.

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. I can reproduce this issue in the NVDA:
![Issue-3334-NVDA](https://user-images.githubusercontent.com/23376742/99064736-6a718700-25b7-11eb-9fa1-27394bf0f129.gif)

But the Narrator works correctly for same case:
![Issue-3334-Narrator](https://user-images.githubusercontent.com/23376742/99064718-60e81f00-25b7-11eb-9d7e-97a5ecf1f15a.gif)

Probably, it is external issue of the NVDA. Should we try to fix it for the NVDA?

**Comment by merriemcgaw** [comment]
@Olina-Zhang  what happens with other Accessibility tools - Inspect and AI4Win?

**Comment by Olina-Zhang** [comment]
@merriemcgaw @SergeySmirnov-Akvelon We cannot repro this issue in .Net 5.0 and .Net 6.0 build using attached app in repro steps or created by us, no exception appears. Please see following gif:
![NotRepro](https://user-images.githubusercontent.com/26474449/101890234-626b2e00-3bdb-11eb-8596-d21973ea205f.gif)

**Comment by SergeySmirnov-Akvelon** [comment]
I was able to reproduce this in the 
[master](https://github.com/dotnet/winforms/commit/703cda7570cb724dac8cdc95e777b890cdce9795) and [5.0](https://github.com/dotnet/winforms/commit/cff5155cf78c06ae6fc3f91447d70e5e87697506) branches using the Inspect

![Issue-4228-inspect](https://user-images.githubusercontent.com/23376742/101900113-fc24e200-3bbf-11eb-8b72-10c677bcec1e.gif)

**Comment by merriemcgaw** [comment]
I'll put this on the backlog so we can check it out later.

**Comment by Zheng-Li01** [comment]
Verified the issue with latest 6.0.100-preview.3.21169.6, the issue has been fixed that no exception pops up as below gif.
![4228GIF](https://user-images.githubusercontent.com/38325459/111967541-22e5a380-8b33-11eb-8fe4-419e28b70a3e.gif)
